xmipy/xmiwrapper.py

```diff
@@ -151,16 +151,52 @@ def get_component_name(self) -> str:
         raise NotImplementedError
 
     def get_input_item_count(self) -> int:
-        raise NotImplementedError
+        count = c_int(0)
+        self.execute_function(self.lib.get_input_item_count, byref(count))
+        return count.value
 
     def get_output_item_count(self) -> int:
-        raise NotImplementedError
+        count = c_int(0)
+        self.execute_function(self.lib.get_output_item_count, byref(count))
+        return count.value
 
     def get_input_var_names(self) -> Tuple[str]:
-        raise NotImplementedError
+        len_address = self.get_constant_int("BMI_LENVARADDRESS")
+        nr_input_vars = self.get_input_item_count()
+        len_names = nr_input_vars * len_address
+        names = create_string_buffer(len_names)
+
+        # get a (1-dim) char array (char*) containing the input variable
+        # names as \x00 terminated sub-strings
+        self.execute_function(self.lib.get_input_var_names, byref(names))
```

Comment:
Is there a reason why you didn't put a high-level wrapper around `get_input_var_names`?

Reply:
> 
> 
> Is there a reason why you didn't put a high-level wrapper around `get_input_var_names`?

the wrapper has been used as discussed